### PR TITLE
Code refactoring

### LINKED ISSUE #89: Some code refactoring may be needed
https://github.com/arshadkazmi42/first-issues/blob/02e560176725002735a91ffd9f8c626452c7c24e/first_timers/first_timers.py#L29-L48

### PR DESCRIPTION
**Pull request by gennadis**
Fixes #89 

Hey there! 
Just wanna help you out with some code improvements.
This is my very first open source contribution :-)

### CODE REVIEW

**Review by arshadkazmi42 — CHANGES_REQUESTED** [review]
Thanks alot for working on this.

But I feel there are few things missing in this after the update.

We actually want to search for issues which are labelled `good-first-issue` and `good first issue`

With these new changes, these seems to be missing, also while calling the function we are not passing these labels, which is expected now.

You can check the code at below two lines.

https://github.com/arshadkazmi42/first-issues/blob/master/first_timers/first_timers.py#L14
https://github.com/arshadkazmi42/first-issues/blob/master/first_timers/first_timers.py#L34

We still need to support this.

**Comment by gennadis** [comment]
Thanks for reply!
Looks like I've forgot to push changes to `run.py`.

File: first_timers/first_timers.py [hunk 1]
```diff
@@ -10,43 +10,39 @@
...
+    return f'https://github.com/{user}/{repo}/issues/{issue_num}'
 
 
-def get_first_timer_issues(days_old=DAYS_OLD):
+def get_first_timer_issues(issue_label: str) -> list[dict]:
     """Fetches the first page of issues with the label first-timers-label
     which are still open.
     """
+    res = requests.get(FIRST_ISSUE_QUERY_URL.format(issue_label))
+    res.raise_for_status()
```
Comment (arshadkazmi42) [inline first_timers/first_timers.py:32]: I think we want to still keep support for handling 403 HTTP code like in below line of code.
We want to return the items what ever has been processed if we get a 403 HTTP code.

https://github.com/arshadkazmi42/first-issues/blob/master/first_timers/first_timers.py#L36-L38

Can you make changes to support that.

**Review by arshadkazmi42 — CHANGES_REQUESTED** [review]
Thank you for the quick fix.

I have added another comment, can you check and make changes for that.

File: first_timers/first_timers.py (hunk 1 above)
Comment (gennadis) [reply first_timers/first_timers.py:32]: I think that with you current code implementation, you get one request for each issue label.
What i suggest is the program will loop through queries in `run.py` and fetched items will be added to `new_issues` list. And if some http error will be raised, you'll still get successfully fetched items.

File: first_timers/run.py [hunk 2]
```diff
@@ -4,50 +4,63 @@
...
-        help='Pass if the DB file should be created.')
-@click.option('--creds-path',
-        prompt='Credentials file',
-        default='',
-        help='File which contains Twitter account credentials. '
-             'Not needed if only saving to DB.')
-@click.option('--debug',
-        is_flag=True,
-        help='Run in debug mode (does not tweet).')
+@click.option(
```
Comment (arshadkazmi42) [inline first_timers/run.py:19]: One small change, I see you have updated all single quotes `'` to double quotes `"` in this file.

Can we change it back to single quotes `'` as other files uses single quotes and we want to keep it consistent.

**Review by arshadkazmi42 — CHANGES_REQUESTED** [review]
One more small change

File: first_timers/run.py (hunk 2 above)
Comment (gennadis) [reply first_timers/run.py]: So sorry! Linted code accidentally.

**Review by arshadkazmi42 — APPROVED** [review]
Looks good. Thank you for working on this.

Lets get this merged :tada:

**Comment by arshadkazmi42** [comment]
@gennadis I just checked, our tests are failing. Can you fix the tests.

**Comment by gennadis** [comment]
> @gennadis I just checked, our tests are failing. Can you fix the tests.
> 
> Screenshot from 2022-02-07 12-22-43

Fixed.

**Review by arshadkazmi42 — APPROVED** [review]
:100:

**Comment by arshadkazmi42** [comment]
Thank you @gennadis 

I will get this deployed with the next release.

**Comment by gennadis** [comment]
> Thank you @gennadis
> 
> I will get this deployed with the next release.

Thank you!
It had been a great pleasure working with you!

**Pull request merged**